packages/dds/tree/src/feature-libraries/sequence-field/compose.ts

```diff
@@ -181,46 +387,84 @@ function composeMark<TNodeChange, TMark extends Mark<TNodeChange>>(
     mark: TMark,
     revision: RevisionTag | undefined,
     composeChild: NodeChangeComposer<TNodeChange>,
+    genId: IdAllocator,
+    moveEffects: MoveEffectTable<TNodeChange>,
 ): TMark {
     if (isSkipMark(mark)) {
         return mark;
     }
 
     const cloned = clone(mark);
-    if (revision !== undefined && mark.type !== "Modify") {
-        (cloned as HasRevisionTag).revision = revision;
+    if (moveEffects.validatedMarks.has(mark)) {
+        moveEffects.validatedMarks.add(cloned);
     }
 
-    if (mark.type !== "MoveIn" && mark.changes !== undefined) {
-        (cloned as HasChanges<TNodeChange>).changes = composeChild([
-            tagChange(mark.changes, revision),
-        ]);
+    assert(!isSkipMark(cloned), "Cloned should be same type as input mark");
+    if (revision !== undefined && cloned.type !== "Modify") {
```

Comment:
I think we also want to check that `cloned.revision === undefined` otherwise we'll end up overwritting it.